Review

madis009 opened the pull request.

madis009 committed:
websocket_client initial commit. (#1)

Co-authored-by: Oleksandr1112 <pakholchako@gmail.com>

madis009 committed:
add new line (#2)

madis009 committed:
added parsing part of json files.

madis009 committed:
connect function enabled.

madis009 committed:
implemented cli for services.json and error_codes.json.

madis009 committed:
TUI implemented.

madis009 committed:
changed color to gray and server connected.

madis009 committed:
removed tabs from tui.

madis009 committed:
parsing data and endpoint list added.

madis009 committed:
refactor code

madis009 committed:
added exception handling.

madis009 committed:
remove unnecessary files.

alordash commented on Cargo.toml:
```diff
@@ -0,0 +1,27 @@
+crossterm = "0.28.1"
+clap = { version = "4.0", features = ["derive"] }
+async-trait = "0.1.50"
+anyhow = "1.0"
```
Let's use only `eyre` crate

alordash commented on src/parser/services.rs:
```diff
@@ -0,0 +1,28 @@
+use std::collections::HashMap;
+
+impl Services {
+    pub fn extract_endpoints(&self) -> (Vec<String>, HashMap<String, (String, u32, Vec<String>)>) {
```
Can you please create model for hash map key `(String, u32, Vec<String>)`?  
I see it appears in a lot of places in code

alordash commented on src/ws/client.rs:
```diff
@@ -0,0 +1,73 @@
+
+#[derive(Serialize)]
+struct WsRequest<T: Serialize> {
+    method: u32,
```
Does `method` field mean "method id"? If so, let's rename it to `method_id`

alordash replied:
Same for args names

madis009 committed:
fixed comments.

madis009 committed:
changed services.json.

madis009 committed:
added casting types.

madis009 committed:
added default values.

madis009 committed:
implemented stream data.

madis009 committed:
delete cofig files

madis009 committed:
changed gitignore file.

amonin7 committed:
add the docs for the validator

dVeon-loch committed:
Minor improvements to readme

PakhomovAlexander committed:
Fix parameters reordering.

Before: parameters were sorted by name and send
to the server. This caused deserialization issues.

After: use HashMap to ignore the order in Vec and
sed parameter according to their definition in endpoints.

YohDeadfall committed:
feat: Two step login (#7)

YohDeadfall committed:
Improved error reporting (#8)

dVeon-loch committed:
feat: Add ws-load-test

- Restructure the repo a bit
- Commit some previously in progress stuff for updating validator, not important at all for now
- Implement ws-simple binary in ws-load-test

dVeon-loch committed:
feat: optimize throughput of ws-simple
- also add some extra cli args/features
- add python graphing script for results comparison
- add some example protocol headers
- update deps

dVeon-loch committed:
chore: Add load test running script
